### PR TITLE
feat: add ValidateSpecWith and Cron.ValidateSpec

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

- Add `ValidateSpecWith(spec, parser)` package-level function for validating with any `ScheduleParser` implementation (custom parsers, pre-configured parsers with hash keys, etc.)
- Add `(*Cron).ValidateSpec(spec)` method for validating using the Cron instance's configured parser, enabling pre-validation before `AddFunc`/`AddJob` calls
- Add comprehensive tests for both new functions (18 test cases covering default parser, seconds parser, optional seconds, hash expressions, descriptors, edge cases)

These complement the existing `ValidateSpec(spec, ...ParseOption)` function (already on main) identified as a [top upstream feature candidate](https://github.com/netresearch/go-cron/blob/main/LEARNINGS.md) from dependent analysis.

## Test plan

- [x] `TestValidateSpecWith` - 8 test cases covering standard, seconds, hash, descriptor parsers
- [x] `TestCronValidateSpec` - 10 test cases covering default, WithSeconds, WithSecondOptional options
- [x] Full test suite passes (`go test ./...`)
- [x] All pre-commit and pre-push hooks pass (fmt, tidy, build, gitleaks, lint, vulncheck)

### CODE REVIEW

File: validate.go [hunk 1]
```diff
@@ -68,6 +68,22 @@ func ValidateSpec(spec string, options ...ParseOption) error {
...
 }
 
+// ValidateSpecWith validates a cron expression using any ScheduleParser implementation.
+// This is useful when you have a custom parser or a pre-configured Parser instance.
+//
+// Example:
+//
+//	// Validate with a custom parser
+//	:= cron.NewParser(cron.Second | cron.Minute | cron.Hour | cron.Dom | cron.Month | cron.Dow | cron.Hash).
+//	    WithHashKey("my-job")
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply validate.go]: Fixed — doc example now correctly uses `parser := cron.NewParser(...)` with proper Go indentation.

File: validate.go [hunk 2]
```diff
@@ -68,6 +68,22 @@ func ValidateSpec(spec string, options ...ParseOption) error {
...
+//	if err := cron.ValidateSpecWith("H * * * * *", parser); err != nil {
+//	    // Handle error
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply validate.go]: Fixed — same fix as above, variable name and indentation corrected.

File: cron.go [hunk 3]
```diff
@@ -514,6 +514,26 @@ func WithMissedGracePeriod(d time.Duration) JobOption {
...
+// Example:
+//
+//	c := cron.New(cron.WithSeconds())
+//	if err := c.ValidateSpec("0 30 * * * *"); err != nil {
+//	    return fmt.Errorf("invalid cron expression: %w", err)
+//	}
+func (c *Cron) ValidateSpec(spec string) error {
+	if c == nil || c.parser == nil {
+		return errors.New("cron: parser is nil; ensure cron.New is not configured with a nil parser")
+	}
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply cron.go]: Fixed — split into two distinct checks: nil receiver returns "called ValidateSpec on nil Cron instance", nil parser returns "parser is nil; ensure cron.New is not configured with a nil parser".

File: .golangci.yml [hunk 4]
```diff
@@ -204,6 +204,12 @@ linters:
         path: cron\.go
         text: "waitgroup"
 
+      # False positive: "parser" in doc comment code example is variable name + function name
+      - linters:
+          - dupword
+        path: validate\.go
+        text: "parser"
+
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply .golangci.yml]: Fixed — removed the dupword lint exclusion. Reworded the doc comment to "Validate with a custom parser instance" to avoid the cross-line false positive.
